### PR TITLE
Refs #10970 - Rails4 master merge

### PR DESCRIPTION
**Pull request by johnpmitsch**
includes host unification changes

### CODE REVIEW

**Comment by johnpmitsch** [comment]
[test]

File: app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb [hunk 1]
```diff
@@ -217,15 +218,14 @@ def consumer_activate
       # Set it before calling find_activation_keys to allow communication with candlepin
       User.current    = User.anonymous_admin
       activation_keys = find_activation_keys
-      foreman_host    = find_foreman_host(activation_keys.first.organization)
-
-      sync_task(::Actions::Katello::System::Destroy, foreman_host.content_host) if foreman_host.try(:content_host)
+      host = Katello::Host::SubscriptionFacet.find_or_create_host(params[:facts]['network.hostname'],
```
Comment (cpeters) [inline app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb]: find_or_create_by(:host => params[:facts]['network.hostname']

Comment (johnpmitsch) [inline app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb]: good find. A general note on this that I discovered today, "old style" hashes are no longer accepted by finder methods in rails 4 and seem not to be working correctly. We should start using either

```
find_or_create_by(host: params[:facts]['network.hostname'])
```
or

```
where(host: params[:facts]['network.hostname']).first_or_create
```
and update existing instances.

More info: [rails doc, scroll to 11.2](http://edgeguides.rubyonrails.org/4_0_release_notes.html) and [related stack overflow article](http://stackoverflow.com/questions/22520274/rails-4-find-or-create-by-method-doesnt-work)

Comment (johnpmitsch) [inline app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb]: ohh this is actually a [new method](https://github.com/Katello/katello/blob/master/app/models/katello/host/subscription_facet.rb#L56) and not a rails method

**Comment by cpeters** [comment]
ACK

**Pull request merged**